frame/support/procedural/src/pallet_error.rs

```diff
@@ -39,7 +39,7 @@ pub fn derive_pallet_error(input: proc_macro::TokenStream) -> proc_macro::TokenS
 			syn::Fields::Unnamed(syn::FieldsUnnamed { unnamed: fields, .. }) => {
 				let maybe_field_tys = fields
 					.iter()
-					.map(|f| generate_field_types(f, &frame_support))
+					.map(|f| generate_field_types(f, frame_support))
```

Comment:
wonder if we pay attention on the size of vars that being moved instead borrowing in so many places.
or we check that performance not affected much by regenerating the weights.

Reply:
We're not actually changing anything here. The & is getting auto-dereferenced by the compiler before it calls the function - the type passed to the function is the same, it's just that now it's not needing to optimise away the unnecessary reference and then immediate dereferencing. The compiled asm would be the same, just slightly less work for the compiler and slightly less for the human to read.